File: src/main/old/bbarena/model/event/EventFlowListener.java
package bbarena.model.event;

/**
 * Implementations of this interface will be notified of EventLine changes
 *
 * @author f.bellentani
 */
public interface EventFlowListener {

	void beforeDoEvent(Event e);

	void afterDoEvent(Event e);

	void beforeUndoEvent(Event e);

	void afterUndoEvent(Event e);

	void eventSizeChanged(int newSize);

    void setEventManager(EventManager eventManager);

}

File: src/main/kotlin/bbarena/model/event/EventFlowListener.kt
package bbarena.model.event

/**
 * Implementations of this interface will be notified of EventLine changes
 *
 * @author f.bellentani
 */
interface EventFlowListener {

    fun beforeDoEvent(e: Event)

    fun afterDoEvent(e: Event)

    fun beforeUndoEvent(e: Event)

    fun afterUndoEvent(e: Event)

    fun eventSizeChanged(newSize: Int)

    fun setEventManager(eventManager: EventManager)

}
